src/backend/commands/subscriptioncmds.c

```diff
@@ -1809,7 +1809,18 @@ DropSubscription(DropSubscriptionStmt *stmt, bool isTopLevel)
 
 				ReplicationSlotNameForTablesync(subid, relid, syncslotname,
 												sizeof(syncslotname));
-				ReplicationSlotDropAtPubNode(wrconn, syncslotname, true);
+
+
+				if (disable_logical_replication_subscribers)
+				{
+					ereport(LOG,
+							(errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+								errmsg("do not drop tablesync slot \"%s\" when disable_logical_replication_subscribers=true", syncslotname)));
```

Comment:
```suggestion
								errmsg("cannot drop tablesync slot \"%s\" when disable_logical_replication_subscribers=true", syncslotname)));
```

Reply:
Did some research in `backend/replication/slot.c`, and some error messages use `cannot`.

Reply:
Good point, just feels weird to start a log message with `do`. Maybe `skip dropping tablesync...`